src/app/clusters/scenes/ExtensionFieldsSets.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        https://urldefense.com/v3/__http://www.apache.org/licenses/LICENSE-2.0__;!!N30Cs7Jr!UgbMbEQ59BIK-1Xslc7QXYm0lQBh92qA3ElecRe1CF_9YhXxbwPOZa6j4plru7B7kCJ7bKQgHxgQrket3-Dnk268sIdA7Qb8$
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ExtensionFieldsSets.h"
+
+namespace chip {
+namespace scenes {
+
+ExtensionFieldsSets::ExtensionFieldsSets()
+{
+    // check if any of the clusters with scene attributes are enabled
+    if (this->kExentesionFieldsSetsSize == 1)
+    {
+        // a size of 1 byte indicates an empty struct, or a struct that only contains :
+        // the on-off cluster
+        // the Mode select cluster
+        // the door lock cluster
+#ifdef ZCL_USING_ON_OFF_CLUSTER_SERVER
+        this->enabledFieldSets.onOff = false;
+
+#elif defined(ZCL_USING_MODE_SELECT_CLUSTER_SERVER)
+        this->enabledFieldSets.currentMode = 0;
+#elif defined(ZCL_USING_DOOR_LOCK_CLUSTER_SERVER)
+        this->enabledFieldSets.lockState = 0;
+#else
+        this->empty = true;
+#endif
```

Comment:
This ties all extension field sets together. This is not scalable. It should be managed by cluster, since the clusters also implement the rules for what fields can be logically ignored.

BTW, the spec PR is not done due to the fact that the ColorControl cluster is missing text that did not carry to ZCL6 yet (and so did not make it into Matter) to make it usable. Will try to fix that very soon.

Reply:
New implementation in place now, independent of the data model